vhdl_lang/src/syntax/separated_list.rs

```diff
@@ -5,39 +5,71 @@
 // Copyright (c) 2023, Olof Kraigher olof.kraigher@gmail.com
 
 use crate::ast::{IdentList, NameList, SeparatedList, WithRef};
-use crate::data::DiagnosticResult;
+use crate::data::{DiagnosticHandler, DiagnosticResult};
 use crate::syntax::common::ParseResult;
 use crate::syntax::names::parse_name;
 use crate::syntax::Kind::Comma;
-use crate::syntax::{Kind, TokenStream};
+use crate::syntax::{kind_str, Kind, TokenAccess, TokenStream};
+use std::fmt::Debug;
+
+/// Skip extraneous tokens of kind `separator`.
+/// When there are any extra tokens of that kind, mark all the positions of these tokens as erroneous
+fn skip_extraneous_tokens(
+    stream: &TokenStream,
+    separator: Kind,
+    diagnostics: &mut dyn DiagnosticHandler,
+) {
+    if let Some(separator_tok) = stream.pop_if_kind(separator) {
+        let start_pos = stream.get_pos(separator_tok);
+        let mut end_pos = start_pos;
+        while let Some(separator_tok) = stream.pop_if_kind(separator) {
+            end_pos = stream.get_pos(separator_tok)
+        }
+        diagnostics.error(
```

Comment:
Did you consider creating one diagnostic per token instead?

Reply:
I didn't so far. This is probably an extreme edge-case so I feel indifferent to whatever solution. Probably a single diagnostic per token would have been easier to implement. Still, I think it's better (and simpler for the user) to have one single error message for all tokens.

Reply:
Its not a big deal. I just think it would make the code simpler. It is not a blocking comment from me.